packages/zoe/src/zoeService/installationStorage.js

```diff
@@ -1,16 +1,23 @@
 import { assert, details as X } from '@agoric/assert';
 import { E } from '@endo/eventual-send';
 import {
+  M,
   makeScalarBigMapStore,
   provideDurableWeakMapStore,
+  vivifyFarInstance,
   vivifyKind,
 } from '@agoric/vat-data';
 import { initEmpty } from '@agoric/store';
+import { InstallationShape } from '../typeGuards.js';
 
 /** @typedef { import('@agoric/swingset-vat').BundleCap} BundleCap */
 /** @typedef { import('@agoric/swingset-vat').BundleID} BundleID */
 /** @typedef {import('@agoric/vat-data').Baggage} Baggage */
 
+export const SourceBundleShape = M.recordOf(M.string(), M.string());
+export const ModuleFormatBundleShape = M.partial({ moduleFormat: M.any() });
+export const BundleShape = M.or(ModuleFormatBundleShape, SourceBundleShape);
```

Comment:
@Chris-Hibbert it seems you've answered the bundle typing question you needed here.

@warner @kriskowal if so, nevermind other request.

Reply:
@kriskowal care to comment?

Until we hear back from @kriskowal , I'll consider this done enough. Resolving.

Reply:
These patterns will allow all valid bundles and many invalid bundles. `const BundleShape = M.recordOf(M.string(), M.string({ maxLength: Infinity }))` would be a bit more specific and also allow all valid bundles. `moduleFormat` must be a string and also must be present in all bundle shapes. Also, everything sensitive to bundle shapes currently does extensive validation without the pattern language, so `M.any()` would also suffice here.

I think the narrowest pattern for bundle source that allows for future evolution of the format is:
```js
export const BundleShape = M.and(M.partial({ moduleFormat: M.string() }), M.recordOf(M.string(), M.string()))
```

Reply:
Hi @kriskowal thanks. @Chris-Hibbert , I'd prefer the more precise pattern for now, even if it is redundant with validation done elsewhere. Unresolving.

Reply:
Looks done. Resolving.